jax/interpreters/batching.py

```diff
@@ -123,22 +123,36 @@ def sublift(self, val):
     return BatchTracer(self, val.val, val.batch_dim)
 
   def get_primitive_batcher(self, primitive):
-    if primitive in initial_style_batchers:
-      return partial(initial_style_batchers[primitive],
-                    axis_name=self.axis_name,
-                    main_type=self.main.trace_type)
     try:
       return primitive_batchers[primitive]
     except KeyError as err:
       msg = "Batching rule for '{}' not implemented"
       raise NotImplementedError(msg.format(primitive)) from err
 
+  def get_axis_primitive_batcher(self, primitive, frame):
```

Comment:
The reason I had to add this method is because it needs to be overridden by `SPMDBatchTrace` in to allow `spmd_primitive_batchers` to have the correct behavior.